app/models/provider_interface/provider_applications_filter.rb

```diff
@@ -160,5 +160,24 @@ def provider_locations_filters
         }
       end
     end
+
+    def subject_filter
+      provider_ids = applied_filters[:provider] || provider_user.providers
+      provider_courses = Course.where(provider: provider_ids)
+
+      {
+        type: :checkbox_filter,
+        heading: 'Search for subject',
+        name: 'subject',
+        options: Subject.joins(:courses).merge(provider_courses).distinct
+        .map do |s|
```

Comment:
@richardpattinson says just call it subject! 🔥

Reply:
@richardpattinson good catch, we really need to set the rubocop rule for this.